packages/libs/src/sdk/services/Auth/UserAuth.ts

```diff
@@ -0,0 +1,149 @@
+import { GetFn, Hedgehog, SetAuthFn, SetUserFn } from '@audius/hedgehog'
+import { keccak_256 } from '@noble/hashes/sha3'
+import * as secp from '@noble/secp256k1'
+import fetch from 'cross-fetch'
+import { EIP712TypedData, MessageData, signTypedData } from 'eth-sig-util'
+
+import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
+
+import { defaultUserAuthConfig } from './constants'
+import type { UserAuthConfig, AuthService } from './types'
+
+export class UserAuth implements AuthService {
+  /**
+   * Configuration passed in by the consumer (with defaults)
+   */
+  private config: UserAuthConfig
+
+  /**
+   * Hedgehog instance to handle auth
+   */
+  private readonly hedgehog: Hedgehog
+
+  constructor(config?: UserAuthConfig) {
+    this.config = mergeConfigWithDefaults(config, defaultUserAuthConfig)
+
+    const get: GetFn = async ({ lookupKey, email, otp }) => {
+      const params = new URLSearchParams({
+        lookupKey,
+        username: email as string
```

Comment:
nit: Can we avoid these casts by typing `get` differently? I'd try `{ lookupKey, email, otp }: Parameters<typeof GetFn>[0] & { email: string; otp: string }` or something like it.

Reply:
we should fix the hedgehog types... because they're unknown. forcing them correctly here via typing this, makes using the `get` fn not fit into the shape GetFn expects